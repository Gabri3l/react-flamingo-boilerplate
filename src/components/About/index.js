import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class About extends Component {
  constructor(props) {
    super(props);
    this.state = {
      test: 'This component has its own state.',
    };
  }

  onButtonClick = () => {
    this.setState({ test: "And it's not styled withe styled components" });
  };

  render() {
    return (
      <div>
        <h1>About</h1>
        <p>{this.state.test}</p>
        <Link to="/">Home</Link>
        <br />
        <br />
        <button onClick={this.onButtonClick}>Change Text</button>
      </div>
    );
  }
}

export default About;
