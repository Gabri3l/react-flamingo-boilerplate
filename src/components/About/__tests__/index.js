import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';

import About from '../../About';

test('About should match snapshot', () => {
  const component = renderer.create(
    <MemoryRouter>
      <About />
    </MemoryRouter>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
