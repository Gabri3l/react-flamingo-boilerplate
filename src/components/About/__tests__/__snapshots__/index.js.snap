// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `
<div>
  <h1>
    About
  </h1>
  <p>
    This component has its own state.
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Change Text
  </button>
</div>
`;
